### PR TITLE
Provide an easy way to create segmented islands

### PR DESCRIPTION
**Pull request by Kemosabert**
### Description
For enhancement request #223 

#### Screenshot after this PR
screen shot 2018-06-20 at 10 57 06

### Breaking changes
none

### CODE REVIEW

File: components/island/theme.css [hunk 1]
```diff
@@ -55,3 +55,34 @@
...
+  &:not(:first-child) {
+    border-right: 0;
+  }
+
+  &:not(:last-child) {
+    border-left: 0;
+  }
+
+  &:first-child {
+    border-radius: 4px 0 0 4px;
```
Comment (nickwaelkens) [inline components/island/theme.css]: Can we move these instances of `4px` to a separate variable (and the one on L6 as well?) Thanks!

**Comment by lowiebenoot** [comment]
I find it a bit weird that the "child" islands are segmented. I would expect that you have to put the `segment` prop on the parent component, which is than more of an `IslandGroup`. Just like we have `ButtonGroup`.

**Comment by Kemosabert** [comment]
@lowiebenoot That is what I initially tried, but a class on the parent did not have an effect on the children with class island, I thought because of how styled components worked. I double checked however and now this does seem to work, so I will make a new commit for this.

**Comment by nickwaelkens** [comment]
I feel like you're almost there, but what @lowiebenoot is trying to say is that we probably need a wrapper component called `IslandGroup` (or `Archipelago` :trollface:) just like we have `Button`s and `ButtonGroup`

**Comment by lowiebenoot** [comment]
Yes, `IslandGroup` can automatically pass the segmented prop down to the child `Island` components.

File: components/island/IslandGroup.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import Box from '../box';
+import theme from './theme.css';
+
```
Comment (nickwaelkens) [inline components/island/IslandGroup.js]: Why pass `color`, but not `dark`? Pass both or don't pass any?

```jsx
const lightColors = ['white', 'neutral'];

isDark () => {
	const { dark, color } = this.props;
	if (lightColors.includes(color)) {
		return false;
	}

	return dark;
}
```

Also, I see this come back again and again. Care to pick this up in a follow-up PR and put this in a util/helper? Thanks!

**Comment by lowiebenoot** [comment]
It would be nice if IslandGroup would pass down more props, just like we do in `ButtonGroup`.

For example, we could set the color prop on the `IslandGroup`.

Now you have to do it like this:
```js
<IslandGroup>
   <Island color="mint">...</Island>
   <Island color="mint">...</Island>
</Island>
```

While it could be like this:
```js
<IslandGroup color="mint">
   <Island>...</Island>
   <Island>...</Island>
</Island>
```

**Comment by Kemosabert** [comment]
@lowiebenoot Completely right, I updated this in the latest commit.

File: components/island/IslandGroup.js (hunk 2 above)
Comment (Kemosabert) [reply components/island/IslandGroup.js]: Yes, I will make a follow-up commit for this.

 :ok: by @Kemosabert on 28/06/2018 10:46:13

**Pull request merged**